generators/app/gulpfile.js

```diff
@@ -10,90 +10,64 @@
- // logging.setVerbose();
-
- const polymer = require('polymer-build');
- const PolymerProject = polymer.PolymerProject;
- const fork = polymer.forkStream;
- const addServiceWorker = polymer.addServiceWorker;
-
- const polymerJSON = require('./polymer.json');
- const project = new PolymerProject(polymerJSON);
-
- // Clean build directory
- gulp.task('clean', () => {
-   return del('build');
- });
-
- gulp.task('build', ['clean'], (cb) => {
-   // process source files in the project
-   const sources = project.sources()
-     .pipe(project.splitHtml())
-     // add compilers or optimizers here!
-     // for example, to process JS files
-     // .pipe(gulpif('**/*.js', babel( // babel settings )))
-     // included is an example demonstrating how to
-     // compress images
-     .pipe(gulpif('**/*.{png,gif,jpg,svg}', imagemin({
-       progressive: true, interlaced: true
-     })))
-     .pipe(project.rejoinHtml());
-
-   // process dependencies (basically the stuff coming out of bower_components)
-   // you can probably ignore these steps but if you want to do something
-   // specific for your installed dependencies, this is the place to do it
-   const dependencies = project.dependencies()
-     .pipe(project.splitHtml())
-      // add code to process your installed dependencies here  
-     .pipe(project.rejoinHtml());
-
-   // merge the source and dependencies streams to we can analyze the project
-   const mergedFiles = mergeStream(sources, dependencies)
-     .pipe(project.analyzer);
-
-   // this fork will vulcanize the project
-   const bundledPhase = fork(mergedFiles)
-     .pipe(project.bundler)
-     // write to the bundled folder
-     .pipe(gulp.dest('build/bundled'));
-
-   const unbundledPhase = fork(mergedFiles)
-     // write to the unbundled folder
-     .pipe(gulp.dest('build/unbundled'));
-
-   cb();
- });
-
- gulp.task('service-worker', ['build'], () => {
-   const swConfig = {
-     navigateFallback: '/index.html',
-   };
-
-   // Once the unbundled build stream is complete, create a service worker for the build
-   const unbundledPostProcessing = addServiceWorker({
-     project: project,
-     buildRoot: 'build/unbundled',
-     swConfig: swConfig,
-     serviceWorkerPath: 'service-worker.js',
-   });
-
-   // Once the bundled build stream is complete, create a service worker for the build
-   const bundledPostProcessing = addServiceWorker({
-     project: project,
-     buildRoot: 'build/bundled',
-     swConfig: swConfig,
-     bundled: true,
-   });
- });
-
- gulp.task('default', ['service-worker']);
+'use strict';
+
+const path = require('path');
+const gulp = require('gulp');
+const gulpif = require('gulp-if');
+const logging = require('plylog');
+
+// Got problems? Try logging 'em
+// logging.setVerbose();
+
+// !!! IMPORTANT !!! //
+// Keep the global.config above any of the gulp-tasks that depend on it
+global.config = {
+  json: path.join(process.cwd(), 'polymer.json'),
+  env: 'prod',
+  src: 'src',
```

Comment:
yeah true. I noticed the `wsk-tasks` use the `env` var to do different things and was thinking about maybe stealing some of their examples...